File: app/src/main/java/org/breezyweather/common/ui/widgets/trend/chart/AbsChartItemView.java
package org.breezyweather.common.ui.widgets.trend.chart;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

public abstract class AbsChartItemView extends View {

    public AbsChartItemView(Context context) {
        super(context);
    }

    public AbsChartItemView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public AbsChartItemView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public abstract int getMarginTop();

    public abstract int getMarginBottom();
}

File: app/src/main/java/org/breezyweather/common/ui/widgets/trend/chart/AbsChartItemView.kt
package org.breezyweather.common.ui.widgets.trend.chart

import android.content.Context
import android.util.AttributeSet
import android.view.View

abstract class AbsChartItemView @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : View(context, attrs, defStyleAttr) {
    abstract val marginTop: Int
    abstract val marginBottom: Int
}
